### ISSUE TITLE
Role arn:minio:iam:::role/dummy-internal does not exist

### ISSUE THREAD

**Issue by oscni** [minio/minio#16468]
When calling the following with curl or postman:
https://[HOSTNAME]?Action=AssumeRoleWithWebIdentity&WebIdentityToken=[JWT]&Version=2011-06-15&DurationSeconds=86400
The following error is returned back:
`<ErrorResponse xmlns="https://sts.amazonaws.com/doc/2011-06-15/"><Error><Type></Type><Code>InvalidParameterValue</Code><Message>Role arn:minio:iam:::role/dummy-internal does not exist</Message></Error><RequestId>173D59BB3832CBD3</RequestId></ErrorResponse>`

JWT seems correct when validating it. ALG is RS256 if is an issue.
Also it works to login to the minio console with OIDC. Console is setup on a slightly different HOSTNAME if that is an issue.

## Expected Behavior
I want to get the XML with access key and secret key

## Current Behavior
Gets an error message

## Steps to Reproduce (for bugs)
Configure Azure AD OIDC app registration
Configure minio with OIDC via web gui

## Your Environment
minio:RELEASES.2023-01-20T02-05-44Z
Kubernetes v1.24.7+rke2r1
Ubuntu OS
Azure AD OIDC app registration

**Comment by harshavardhana** [comment]
> `<ErrorResponse xmlns="https://sts.amazonaws.com/doc/2011-06-15/"><Error><Type></Type><Code>InvalidParameterValue</Code><Message>Role arn:minio:iam:::role/dummy-internal does not exist</Message></Error><RequestId>173D59BB3832CBD3</RequestId></ErrorResponse>`

Do you have region set @oscni ?

**Comment by tom-010** [comment]
I get the same error. Even when I set the region when creating the Minio client. It works on localhost, but not on my server.

**Comment by tom-010** [comment]
I had to delete the docker-volume and re-create it. Now it works...

**Comment by WeiXie96** [comment]
Have the same problem, any suggestions to resolve it?
I call the following with postman: https://localhost:9000/?Action=AssumeRoleWithWebIdentity&WebIdentityToken='the id token from OIDC provider'&Version=2011-06-15&DurationSeconds=3600&Region=us-east-1 
 
And get:
```
    <Error>
        <Type></Type>
        <Code>InvalidParameterValue</Code>
        <Message>Role arn:minio:iam:::role/dummy-internal does not exist</Message>
    </Error>
    <RequestId>1750892C1F3AB10E</RequestId>
</ErrorResponse>
```

**Comment by hemstreet** [comment]
@WeiXie96 did you ever get past that error? I'm hitting it now and wondering if/what you did to solve it

`curl --location --request POST 'https://minio.example.com/?Action=AssumeRoleWithWebIdentity&WebIdentityToken=ey...&Version=2011-06-15`

And i'm getting 
```
<?xml version="1.0" encoding="UTF-8"?>
<ErrorResponse xmlns="https://sts.amazonaws.com/doc/2011-06-15/">
    <Error>
        <Type></Type>
        <Code>InvalidParameterValue</Code>
        <Message>Role arn:minio:iam:::role/dummy-internal does not exist</Message>
    </Error>
    <RequestId>1775E02E71963041</RequestId>
</ErrorResponse>
```

**Comment by sheromon** [comment]
Same problem here using MinIO version RELEASE.2023-05-18T00-05-36Z, calling AssumeRoleWithWebIdentity and getting this response: `<ErrorResponse xmlns="https://sts.amazonaws.com/doc/2011-06-15/"><Error><Type></Type><Code>InvalidParameterValue</Code><Message>Role arn:minio:iam:::role/dummy-internal does not exist</Message></Error><RequestId>177742A4B4B830BB</RequestId></ErrorResponse>`.

**Comment by sheromon** [comment]
I found my problem. I had removed the configuration setting these EVs: `MINIO_IDENTITY_OPENID_CONFIG_URL`, `MINIO_IDENTITY_OPENID_CLIENT_ID`. I set them to the correct values for my system, and now AssumeRoleWithWebIdentity is working fine.

**Comment by space192** [comment]
Hey I have the exact same problem, with google as my openID and I'm using Rolepolicy to manage the accounts, but i still get `<ErrorResponse xmlns="https://sts.amazonaws.com/doc/2011-06-15/"><Error><Type></Type><Code>InvalidParameterValue</Code><Message>Role arn:minio:iam:::role/dummy-internal does not exist</Message></Error><RequestId>179EFDCBB3F4A46E</RequestId></ErrorResponse>` as anyone solved this ?

**Comment by Sjoerd82** [comment]
Getting this as well, with Auth0 using MinIO RELEASE.2024-02-26T09-33-48Z (as Docker). It's difficult to troubleshoot, because I don't understand the error message. Also, I don't have any `MINIO_IDENTITY_*` environment variable.

**Comment by sheromon** [comment]
Yeah, unfortunately, that error message is not very helpful. What I've come to find is that there could be any of a large number of things that could cause this error message. In my case, usually the problems are something revolving around my OpenID integration is not configured correctly. I was using the configuration method where you set the environment variables `MINIO_IDENTITY_OPENID_CONFIG_URL` and others.

Because there are so many things that could be causing this message, there can't really be a single solution to fix this error message.

I have found that the MinIO staff have been helpful to me in the past in their Slack. I've posted questions there and gotten responses from MinIO engineers surprisingly quickly. I would recommend that approach to get help.
